wremnants/theory_tools.py

```diff
@@ -318,6 +332,24 @@ def make_theory_corr_hists(df, name, axes, cols, helpers, generators, modify_cen
         unc = df.HistoBoost(hist_name, axes, [*cols, f"{generator}Weight_tensor"], tensor_axes=helpers[generator].tensor_axes[-1:], storage=hist.storage.Double())
         res.append(unc)
 
+        var_axis = helpers[generator].tensor_axes[-1]
+
+        # special treatment for Omega since it needs to be decorrelated in charge and rapidity
+        if "Omega0." in var_axis:
```

Comment:
Maybe better to check any("Omega" in x for x in var_axis) since nothing forces you to have Omega = 0. as the variation

Reply:
Updated this to be more general